proxy/http/HttpUserAgent.cc

```diff
@@ -0,0 +1,63 @@
+/** @file
+
+  A brief file description
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include "HttpUserAgent.h"
+
+#include "I_IOBuffer.h"
+#include "ProxyTransaction.h"
+
+HttpVCTableEntry *
+HttpUserAgent::get_entry() const
+{
+  return m_entry;
+}
+
+void
+HttpUserAgent::set_entry(HttpVCTableEntry *entry)
+{
+  m_entry = entry;
+}
+
+IOBufferReader *
+HttpUserAgent::get_raw_buffer_reader()
+{
+  return m_raw_buffer_reader;
+}
+
+void
+HttpUserAgent::set_raw_buffer_reader(IOBufferReader *raw_buffer_reader)
+{
+  m_raw_buffer_reader = raw_buffer_reader;
+}
+
+ProxyTransaction *
+HttpUserAgent::get_txn() const
+{
+  return m_txn;
+}
+
+void
+HttpUserAgent::set_txn(ProxyTransaction *txn)
+{
+  m_txn = txn;
+}
```

Comment:
This is pretty hot code. We probably need to keep these inlined in the header file. (Declare inline, and provide definitions at the bottom of the header file.)